doc/guide/glossary.rst

```diff
@@ -4,13 +4,116 @@ Glossary
 Here we define Flux-specific and general HPC and workload management terms
 used in our documentation that may not be familiar to all readers.
 
-  .. glossary::
+.. glossary::
 
-     initial program
-        A user-defined program, such as a batch script, launched on the first
-	node of a Flux instance.  Its purpose is to launch and monitor a
-	workload.  Once it is complete, the instance exits.
+  enclosing instance
+    The Flux instance that launched the current Flux instance as a job.
```

Comment:
Took me a second to grok, perhaps "The parent Flux instance that launched ..."?

Reply:
I think we've also used enclosing instance to refer to the nearest instance under which a process is running, e.g. that to which `FLUX_URI` refers, or the system instance if the system is running Flux as the native RM. Am I mistaken?

Applying the term only to instances run as a job under Flux may reduce any ambiguity, but IMO it is too convenient to use _enclosing instance_ also to refer generally to the current instance under which a process is running.

If we do want to restrict the usage of the term, there may be a few spots in existing documentation that need to be tightened up.

Reply:
> I think we've also used enclosing instance to refer to the nearest instance under which a process is running

Now that @grondo mentions it, I have this feeling as well.

Reply:
Sorry I got off track on that one!  How's this?

> The Flux instance that a process naturally interacts with. It is the instance referred to by the FLUX_URI environment variable, or if that is not set, it is the system instance.